sdk/ml/azure-ai-ml/azure/ai/ml/entities/_credentials.py

```diff
@@ -606,36 +606,37 @@ def _from_online_endpoint_rest_object(cls, obj: RestManagedServiceIdentityConfig
 
     @classmethod
     def _from_workspace_rest_object(cls, obj: RestManagedServiceIdentityConfiguration) -> "IdentityConfiguration":
-        user_assigned_identities = None
-        if obj.user_assigned_identities:
-            user_assigned_identities = {}
-            for k, v in obj.user_assigned_identities.items():
-                metadata = None
-                if v and isinstance(v, RestUserAssignedIdentity):
-                    metadata = ManagedIdentityConfiguration._from_workspace_rest_object(
-                        v
-                    )  # pylint: disable=protected-access
-                user_assigned_identities[k] = metadata
-        return cls(
-            type=obj.type,
-            principal_id=obj.principal_id,
-            tenant_id=obj.tenant_id,
-            user_assigned_identities=user_assigned_identities,
+        from_rest_user_assigned_identities = (
+            [
+                ManagedIdentityConfiguration._from_identity_configuration_rest_object(uai, resource_id=resource_id)
+                for (resource_id, uai) in obj.user_assigned_identities.items()
+            ]
+            if obj.user_assigned_identities
+            else None
         )
+        result = cls(
+            type=camel_to_snake(obj.type),
+            user_assigned_identities=from_rest_user_assigned_identities,
+        )
+        result.principal_id = obj.principal_id
+        result.tenant_id = obj.tenant_id
+        return result
 
-    def _to_workspace_rest_object(self) -> RestManagedServiceIdentityConfiguration:
-
-        user_assigned_identities = (
+    def _to_workspace_rest_object(self, existing_workspace = None) -> RestManagedServiceIdentityConfiguration:
+        rest_user_assigned_identities = (
             {uai.resource_id: uai._to_workspace_rest_object() for uai in self.user_assigned_identities}
             if self.user_assigned_identities
             else None
         )
-
+        # add the uai resource_id which needs to be deleted (which is not provided in the list)
+        if existing_workspace and existing_workspace.identity and existing_workspace.identity.user_assigned_identities:
```

Comment:
Can we please keep this logic in workspace operations? The class is meant for translation alone.

Reply:
> 

sure, fixed.